python_modules/libraries/dagster-deltalake/dagster_deltalake/asset.py

```diff
@@ -0,0 +1,17 @@
+from dagster import DataVersion, observable_source_asset
+
+from .resource import DeltaTableResource
+
+"""
+Running the following code on an existing delta table gets you the last updated timestamp:
+deltaTable.history(1).select("timestamp").collect()[0][0]
+"""
+
+
+@observable_source_asset
+def observe_table_last_modified_time(context, delta_table: DeltaTableResource) -> DataVersion:
+    table = delta_table.load()
+    timestamp = table.history(1)[0]["timestamp"]
```

Comment:
This might fail. The commitInfo is entirely optional in the delta protocol. You could rather use the table version instead and make that timestamp optional

Reply:
Ah, interesting. Thanks for the catch. 
This PR is mostly an RFC, in a landed version, we won't be using data version, but rather some sort of metaadata about the observation. The goal here is to represent freshness based on some timestamp, so i don't think table version would work for that goal. 
Going to do some digging on my own, but do you know of the conditions when this information will / won't be available?